chainer/training/extensions/evaluator.py

```diff
@@ -0,0 +1,168 @@
+import copy
+
+import six
+
+from chainer.dataset import convert
+from chainer.dataset import iterator as iterator_module
+from chainer import link
+from chainer import reporter as reporter_module
+from chainer.training import extension
+from chainer import variable
+
+
+class Evaluator(extension.Extension):
+
+    """Trainer extension to evaluate models on a validation set.
+
+    This extension evaluates the current models by a given evaluation function.
+    It creates a :class:`~chainer.Reporter` object to store values observed in
+    the evaluation function on each iteration. The report for all iterations
+    are aggregated to :class:`~chainer.DictSummary`. The collected mean values
+    are further reported to the reporter object of the trainer, where the name
+    of each observation is prefixed by the evaluator name. It also report their
+    standard deviations, where the names are also suffixed by ``.std``.
+
+    Evaluator has a structure to customize similar to that of
+    :class:`~chainer.training.StandardUpdater`. The main differences are:
+
+    - There are no optimizers in an evaluator. Instead, it holds links
+      to evaluate.
+    - An evaluation loop function is used instead of an update function.
+    - Preparation routine can be customized, which is called before each
+      evaluation. It can be used, e.g., to initialize the state of stateful
+      recurrent networks.
```

Comment:
You can, but currently it does not have *post* hook, so you cannot change it to the training mode after the evaluation. Thus, it is recommended to copy the model in that case.